packages/x-date-pickers/src/DesktopDatePicker/tests/DesktopDatePicker.test.tsx

```diff
@@ -307,15 +307,6 @@ describe('<DesktopDatePicker />', () => {
 
       expect(screen.getByLabelText('Next month')).not.to.have.attribute('disabled');
     });
-
-    it('should allow to navigate to previous and next month if props.minDate == null', () => {
```

Comment:
According to the typing `minDate` should not accept `null`
I think this test is an old one that existed before the `describeValidation`, with this PR the typing now complains when you try to pass `null` to this prop and I think it should.

Do you think we should start accepting `null` for this prop? (and the related ones)

Reply:
> Do you think we should start accepting `null` for this prop? (and the related ones)

If it did accept `null` before and now it doesn't, then I only mentioned that it is a BC that we need to act accordingly on.
Accepting `null` makes no sense to me. 🤷 
If someone needs to, they can just pass `undefined` in case of conditional validation value. 🤔

Reply:
It did not accept `null` in the sense that if you did `<DatePicker<Dayjs> minDate={null} />` is was correctly complaining.
Even if you only did `<DatePicker value={value} minDate={null} />` with `value` of type `Dayjs | null`, then the picker would correctly infer that if `value` is `Dayjs | null | undefined`, then `TDate` is `Dayjs` and therefore `minDate` is `Dayjs | undefined`

But if you only pass `minDate`, it infers `TDate` from `minDate` and does it wrong before this PR

Maybe in the migration guide we can just do a quick summary of the type accepted by type of props.
The usecase of this test is not super realistic though, most people are either passing `value` on `defaultValue`, passing neither is not super common

Reply:
Fair points. It was working, but somewhat incorrectly, indeed. 🤔 
Given this discussion, I'd be fine treating it as a bugfix, but maybe one that is does not need to be backported to v6? 🤔

Reply:
I agree not to backport it because of the potential breaking aspect and the complexity of the backport :ok_hand: